File: examples/canonical-iac-pulumi/dotnet/azure/components/secret-component/SecretComponent.cs
using Pulumi;
using Pulumi.AzureNative.KeyVault;

public sealed class SecretComponent : ComponentResource
{
    public Output<string> SecretName { get; private set; }

    public SecretComponent(string name, string environment, string namePrefix, ComponentResourceOptions? opts = null)
        : base("accb:azure:SecretComponent", name, opts)
    {
        var stackName = Deployment.Instance.StackName;
        var vault = new Vault($"{namePrefix}-{environment}-{stackName}-kv", new(), new CustomResourceOptions { Parent = this });
        SecretName = vault.Name.Apply(v => $"{v}-{environment}-config");
        RegisterOutputs(new Dictionary<string, object?> { ["secretName"] = SecretName });
    }
}

File: examples/canonical-iac-pulumi/python/aws/components/secret_component.py
import pulumi


class SecretComponent(pulumi.ComponentResource):
    def __init__(self, name: str, environment: str, name_prefix: str, opts=None):
        super().__init__("accb:aws:SecretComponent", name, None, opts)
        stack = pulumi.get_stack()
        self.secret_ref = pulumi.Output.from_input(f"{name_prefix}-{environment}-{stack}-config")
        self.register_outputs({"secret_ref": self.secret_ref})
